modules/Utils.js

```diff
@@ -206,11 +206,7 @@ class Utils {
         if(metadata == null) return null;
         for(const video of metadata) {
             if(video.video_url === video_url) {
-                if(playlist_url == null) {
-                    return video;
-                } else if(playlist_url === video.playlist_url) {
-                    return video;
-                }
+                return video;
```

Comment:
actually those changes break 2 new tests, can you tell me why they are required? then we can adjust the expectation or remove the tests

Reply:
Yes, without the additional return statement the playlist metadata is not made available at the time it is consumed by the individual threads that are downloading the video. I have no idea why there was a conditional statement there at all as it appears to have no effect. Of course there is no documentation or comments to refer to..

You can try removing the return statement and you will see the old behavior repeat itself. None of the tests failed for me,  can you refer to which tests are failing so I can investigate further?

Thanks!